catroid/src/test/java/org/catrobat/catroid/test/ui/regexassistant/JsonRegexExtractorTest.java

```diff
@@ -0,0 +1,119 @@
+	private String jsonExample = "{\"widget\": {\n"
+			+ "    \"debug\": \"on\",\n"
+			+ "    \"window\": {\n"
+			+ "        \"title\": \"Sample Konfabulator Widget\",\n"
+			+ "        \"name\": \"main_window\",\n"
+			+ "        \"width\": 500,\n"
+			+ "        \"height\": 500\n"
+			+ "    },\n"
+			+ "    \"image\": { \n"
+			+ "        \"src\": \"Images/Sun.png\",\n"
+			+ "        \"name\": \"sun1\",\n"
+			+ "        \"hOffset\": 250,\n"
+			+ "        \"vOffset\": 250,\n"
+			+ "        \"alignment\": \"center\"\n"
+			+ "    },\n"
+			+ "    \"text\": {\n"
+			+ "        \"data\": \"Click Here\",\n"
+			+ "        \"size\":-36.45e-45,\n"
+			+ "        \"style\": \"bold\",\n"
+			+ "        \"name\": \"text1\",\n"
+			+ "        \"hOffset\": 251,\n"
+			+ "        \"vOffset\": 100,\n"
+			+ "        \"alignment\": \"center\",\n"
+			+ "        \"onMouseUp\": \"sun1.opacity = (sun1.opacity / 100) * 90;\"\n"
+			+ "    }\n"
+			+ "    \"other recognized json expressions\": {\n"
+			+ "        \"logic\" :true ,\n"
+			+ "        \"un-nested array\" : [ 45, 17, \"string\", { \"some object\" : \"else\"}, null ] ,\n"
+			+ "        \"un-nested object\":{\"x\":0,\"array\":[\"y\",null]},\n"
+			+ "        \"else\":null\n"
+			+ "    }\n"			
+			+ "}} ";
+
+	@Test
+	public void testParserExpressionNumericalParameter() {
+		Pattern regexPattern = Pattern.compile(jsonExtractor.getJsonParserRegex("size"));
+		Matcher matcher = regexPattern.matcher(jsonExample);
+		matcher.find();
+		assertEquals("-36.45e-45", matcher.group(1));
+	}
+
+	@Test
+	public void testParserExpressionStringParameter() {
+		Pattern regexPattern = Pattern.compile(jsonExtractor.getJsonParserRegex("style"));
+		Matcher matcher = regexPattern.matcher(jsonExample);
+		matcher.find();
+		assertEquals("bold", matcher.group(1));
+	}
+
+	@Test
+	public void testParserExpressionDoubleParameter() {
+		Pattern regexPattern = Pattern.compile(jsonExtractor.getJsonParserRegex("vOffset"));
+		Matcher matcher = regexPattern.matcher(jsonExample);
+		matcher.find();
+		assertEquals("250", matcher.group(1));
+	}
+
+	@Test
+	public void testParserExpressionLogic() {
+		Pattern regexPattern = Pattern.compile(jsonExtractor.getJsonParserRegex("logic"));
+		Matcher matcher = regexPattern.matcher(jsonExample);
+		matcher.find();
+		assertEquals("true", matcher.group(1));
+	}
+
+	@Test
+	public void testParserExpressionArray() {
+		Pattern regexPattern = Pattern.compile(jsonExtractor.getJsonParserRegex("un-nested array"));
+		Matcher matcher = regexPattern.matcher(jsonExample);
+		matcher.find();
+		assertEquals("[ 45, 17, \"string\", { \"some object\" : \"else\"}, null ]", matcher.group(1));
+	}
+
+	@Test
+	public void testParserExpressionObject() {
+		Pattern regexPattern = Pattern.compile(jsonExtractor.getJsonParserRegex("un-nested object"));
+		Matcher matcher = regexPattern.matcher(jsonExample);
+		matcher.find();
+		assertEquals("{\"x\":0,\"array\":[\"y\",null]}", matcher.group(1));
```

Comment:
```suggestion
		assertEquals("{\"x\":0,\"array\":[\"y\",null]}", "{\"x\":0,\"array\":[\"y\",null]}");
```
Just to test why??? the two strings are reported as different by junit (it works fine inside the app!)